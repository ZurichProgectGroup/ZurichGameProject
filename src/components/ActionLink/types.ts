import { ReactNode, MouseEvent } from 'react';

export type OwnProps = {
    className?: string,
    children: ReactNode,
    onClick: (event: MouseEvent<HTMLElement>) => void
};

export type Props = OwnProps;
