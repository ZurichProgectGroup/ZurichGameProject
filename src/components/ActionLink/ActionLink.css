@import "Components/LinkButton/LinkButton.css";
