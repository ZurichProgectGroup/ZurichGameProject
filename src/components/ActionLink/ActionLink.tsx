import React from 'react';
import './ActionLink.css';
import type { Props } from './types';

const ActionLink = ({
    className = 'link-button',
    children,
    ...linkProps
}
: Props) => (
    <a className={className} {...linkProps}>
        {children}
    </a>
);

export default ActionLink;
