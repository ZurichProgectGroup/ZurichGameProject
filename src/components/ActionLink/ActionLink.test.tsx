import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ActionLink from './ActionLink';

it('renders correctly', () => {
    const { asFragment } = render(<ActionLink onClick={() => {}}>test link</ActionLink>);

    expect(asFragment()).toMatchSnapshot();
});

it('calls callback on click', () => {
    const mockCallback = jest.fn();

    const { getByText } = render(<ActionLink onClick={mockCallback}>link</ActionLink>);

    fireEvent.click(getByText('link'));

    expect(mockCallback).toHaveBeenCalledTimes(1);
});
