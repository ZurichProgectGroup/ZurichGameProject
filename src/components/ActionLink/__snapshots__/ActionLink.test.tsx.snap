// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <a
    class="link-button"
  >
    test link
  </a>
</DocumentFragment>
`;
